packages/scenario/src/jr/event/PositionalEvent.ts

```diff
@@ -0,0 +1,85 @@
+import { assertInstanceType } from '@odk-web-forms/common/lib/runtime-types/instance-predicates.ts';
+import type {
+	GroupNode,
+	RepeatInstanceNode,
+	RepeatRangeNode,
+	RootNode,
+	SelectNode,
+	StringInputNode,
+} from '../../client/node-types.ts';
+import type { Scenario } from '../Scenario.ts';
+
+export interface PositionalEventTypeMapping {
+	readonly BEGINNING_OF_FORM: RootNode;
+	readonly QUESTION: SelectNode | StringInputNode;
+	readonly GROUP: GroupNode;
+	readonly REPEAT: RepeatInstanceNode;
+	readonly REPEAT_JUNCTURE: never; // per @lognaturel: this can be ignored
```

Comment:
To elaborate a little bit: as far as I know, there was at some point a way to specify a different UI for adding repeats that used this event type. I think it was for something like a chat interface.

Reply:
I think that's possible but my memory is fuzzy on that. Inserting repeat instances at arbitrary indexes is not currently exposed in Collect. It does get requested now and then and could certainly be valuable in a more table-like display of repeat instances. That would be closer to the original W3C XForms repeat concept and what [Orbeon](https://doc.orbeon.com/xforms/actions/repeat-insert-delete#using-a-trigger-to-execute-actions) and [Fore](https://jinntec.github.io/Fore/demo/todo2.html) expose.